design/datamover.md

```diff
@@ -0,0 +1,48 @@
+# Add Datamover to Velero
+## Problem Statement
+Today, Velero supports snapshotting of volumes using a supported plugin. For some volume/plugin types, these snapshots do not provide durability and portability. Velero would like to implement a generic data mover interface that allows vendors to supply their own data mover to be used for compatible storage types.
+- Specifically, CSI snapshots are not guaranteed to be durable or portable, and this is preventing the CSI plugin from going GA.
+- Kubernetes is moving in such a way that every volume would be backed by a CSI driver. CSI snapshotting allows for a way for a user to take a snapshot, but does not guarantee durability nor portability. We want to provide a generic way for users to guarantee portability and durability of these snapshots for a given provider.
+- We also want to allow vendors to provide their own volume data management strategies to be able to work within the velero backup/restore workflow. 
+
+## What is a Data Mover ?
+Data Mover defines an interface that enables movement of kubernetes data between backup storage.
+
+## Goals
+- The data mover should be **Pluggable**. This means that a user can leverage different types of movers for the same set of resources as long as the movers conform to a similar interface.
+- The interface for Data Mover should support moving data associated with resources **other than** Persistent Volumes
+  - Persistent Volumes are an immediate need, and will likely drive most of the original design discussions.
+  - Ex. a Data Mover for Kubevirt would move VM data to backup storage. The interface should be generic enough to support this use case.
+- The interface for Data Mover should enable publishing it’s specific configuration parameters for users to have fine-grained control over the movement workflow
+  - Examples:
+    - Data mover for vsphere volume can configure block vs file mount usage
+    - Kubevirt mover can configure export file type
+    - Image mover can specify image format type
+- End-user interaction with data mover should be exposed through the existing backup/restore operations
+
+## Non-Goals
+- This implementation will **NOT** change the existing Velero backup/restore workflow.
+  - It is possible in the future that this interface can be used to replace some legacy Velero implementations, but that is **NOT** a goal for this design/implementation.
+- This design of the interface should not include any specific details about (aside from samples):
+  - A mover type
+    - Although we are focused on volumes for phase 1, the interface should have zero knowledge of the underlying data that needs to be moved.
+  - Mover implementations
+    - Ex. How a mover handles volumes at the block vs filesystem level is not relevant to the overarching interface design
+
+## Use Cases
```

Comment:
We should also address the scenario,  to make sure that when a backup is deleted, the moved data is also deleted

Reply:
we may also need to address the use case where vendors might want to create a copy of data in the vendor tenant rather than the customer tenant (hardening) - would help in cases where customer's cluster/tenant is compromised (ex: ransomware). In this case vendor can perhaps use the DataMoverBackup CR to explicitly given an intent to harden the data (specifying a target storage SAS which has some expiry). However, deletion would be something vendor might directly control and may want the delete on DataMoverBackup CR to fail.